return an error status on error in rosbag

phil-marble opened the pull request:
I added some sys.exit(1)'s to rosbag commands to help with scripting.
Specifically, I expected a call to `rosbag info test.bag.active` to
return an error when the bag needed re-indexing.

fixes #1256

phil-marble commented:
Hey there.. New here and interested in getting this fix into kinetic...

wjwwood commented:
@phil-marble it typically takes a few days for someone to have time to review and either merge this or give you feedback on what needs to change. After merging, it will take a few weeks to a month to get into the public `.deb` files. It might take slightly longer during this part of the year due to people traveling for the holidays.

phil-marble commented:
Thanks @wjwwood ! It's not a huge rush, just wondering about process.

I submitted this PR against the default branch (`lunar-devel`.) How does the org manage cherry-picking to older versions?

wjwwood reviewed (commented):
The change looks reasonable to me.

I only identified this line which might also benefit from using `sys.exit` instead of `return`:

https://github.com/phil-marble/ros_comm/blob/c7aee3043fc84fc80b6aae31e5dc7bf4d7a72d53/tools/rosbag/src/rosbag/rosbag_main.py#L298

I think `sys.exit` make more sense because of how the "cmd" functions are executed:

https://github.com/phil-marble/ros_comm/blob/c7aee3043fc84fc80b6aae31e5dc7bf4d7a72d53/tools/rosbag/src/rosbag/rosbag_main.py#L900

While this line (and few others that follow it) could (probably should) be converted to use `sys.exit` rather than `exit`:

https://github.com/phil-marble/ros_comm/blob/c7aee3043fc84fc80b6aae31e5dc7bf4d7a72d53/tools/rosbag/src/rosbag/rosbag_main.py#L493

After a quick scan I didn't see anything else that needed doing related to returning an error code when there's a problem.

I'll leave it up to one of the actual maintainers to review and approve this change however.

wjwwood commented:
Opening against lunar-devel (or the latest branch) is the right thing to do. The maintainers will back port changes at their discretion or your request.

dirk-thomas commented on tools/rosbag/src/rosbag/rosbag_main.py:
```diff
@@ -449,6 +452,7 @@ def fix_cmd(argv):
                     
         print('Try running \'rosbag check\' to create the necessary rule files or run \'rosbag fix\' with the \'--force\' option.')
         os.remove(outname)
+	sys.exit(1)
```
The indentation of this line uses a tab which isn't allowed in Python 3. Please change this to spaces.

mikepurvis commented:
:+1:

phil-marble committed:
return an error status on error in rosbag

I added some sys.exit(1)'s to rosbag commands to help with scripting.
Specifically, I expected a call to `rosbag info test.bag.active` to
return an error when the bag needed re-indexing.

phil-marble commented:
removed the tab

dirk-thomas commented:
Thank you.
